File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/ClientRecordListWrapper.java
package kz.greetgo.sandbox.controller.model;

import java.util.List;

// TODO: 02.10.18 для второго пэйджа count не нужен

public class ClientRecordListWrapper {
  public List<ClientRecord> records;
  public int count;

  public ClientRecordListWrapper(List<ClientRecord> records, int count) {
    this.records = records;
    this.count = count;
  }

  public ClientRecordListWrapper() {
  }
}

File: sandbox.client/src/model/ClientRecordListWrapper.ts
import {ClientRecord} from "./ClientRecord";

export class ClientRecordListWrapper {
  public records: ClientRecord[];
  public count: number/*int*/;

  public static create(a: any): ClientRecordListWrapper {
    const ret = new ClientRecordListWrapper();
    ret.assign(a);
    return ret;
  }

  assign(a: any) {
    this.records = a.records;
    this.count = a.count;
  }
}
